src/bootstrap.js

```diff
@@ -536,18 +546,21 @@ module.exports = (ceConfig = {}, pnConfig = {}) => {
      * @fires $"."connected
      */
     ChatEngine.connect = (uuid, state = {}, authKey) => {
-
-        // this creates a user known as Me and
-        // connects to the global chatroom
-        ChatEngine.pnConfig.uuid = uuid;
-        ChatEngine.pnConfig.authKey = authKey;
-
-        if (authKey) {
-            ChatEngine.handshake(() => {
+        if (!ChatEngine.isConnected) {
+            // this creates a user known as Me and
+            // connects to the global chatroom
+            ChatEngine.pnConfig.uuid = uuid;
+            ChatEngine.pnConfig.authKey = authKey;
+
+            if (authKey) {
+                ChatEngine.handshake(() => {
+                    ChatEngine.firstConnect(state);
+                });
+            } else {
                 ChatEngine.firstConnect(state);
-            });
-        } else {
-            ChatEngine.firstConnect(state);
+            }
+
+            ChatEngine.isConnected = true;
```

Comment:
@ManuelFernando this looks like a bit too soon for case, when `authKey` provided, because client will be actually connected only after handshake completion.
So, it looks ok, to place it in _else_ branch and put right after `ChatEngine.firstConnect(state);` inside of `handshake` completion handler.